### DISCUSSION TITLE
Consumer to Provider payload

### DISCUSSION THREAD

**Discussion by Jayasheel** [eclipse-edc/Connector#1613] [Q&A] [answered]
All,

I am looking to attach a payload from consumer to provider and then use this payload to make a REST call from provider to a REST API server. The results from the REST Call are returned back to the consumer.

![integration](https://user-images.githubusercontent.com/8556923/177975396-94857010-a2be-4eb8-885f-782c297fbb59.png)

I used sample 04.0 to and modified provider to make a REST GET call to API server, I need to add the payload from consumer and also make a POST call from provider to API server.

Any pointers and help is much appreciated!

Thanks
Jayasheel

**Comment by hlange** [comment, marked as answer]
This sounds similar to my discussion (https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/discussions/1386).

@jimmarino has made some suggestions how this can be implemented as short term solution and created also an issue (https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/1420) for this.

**Reply by Jayasheel** [reply]
Thank you @hlange For now have worked around this.

**Reply by waleedshabbir1** [reply]
@Jayasheel Could I ask what kind of work around you implemented for this? I am in the same situation

**Reply by dilip-dhankecha** [reply]
@Jayasheel Is there any update on this one?
or do you have an example for the same or not?

**Comment by hlange** [comment]
I wrote something that could help: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/discussions/2107#discussioncomment-4224961
